### ISSUE TITLE
could not find `linkme` in the list of imported crates

### ISSUE THREAD

**Issue by PonasKovas** [dtolnay/linkme#108]
Hello, thanks for another great crate!

I am trying to wrap this functionality in my own crate, but I am getting this error when trying to use it:
```rust
error[E0433]: failed to resolve: could not find `linkme` in the list of imported crates
```

This is because here I use my own "wrapper" crate and don't import `linkme` directly. I don't want to burden my users with importing internal crates like this.

I assume this happens because your macro needs to access `linkme::DistributedSlice` type to construct it.

Would you mind adding an option to give the path to the `linkme` crate in the macro invocation to solve this issue?

I think something like this would work well:
```rust
#[distributed_slice(@my::reexported::path::to::linkme)]
pub static BENCHMARKS: [fn(&mut Bencher)];
#[distributed_slice(@my::reexported::path::to::linkme, BENCHMARKS)]
fn bench_deserialize(b: &mut Bencher) {}
```

Basically adding an optional argument that starts with `@` in the `distributed_slice` macro to give a path to `linkme`.

If this is fine to you, **I will make a PR**.

Wishing you a blessed Christmas and a joyful New Year, filled with the peace and love of Christ.

**Comment by dtolnay** [comment]
I would prefer not to support such a workaround in this crate. But in case you are interested in working on this instead, https://github.com/rust-lang/rust/issues/54363#issuecomment-1237544924 describes what I think is a good design for proc macro re-exporting.
